### ISSUE TITLE
[BUG] Unable to reach cellxgene servers

### ISSUE THREAD

**Issue by LysSanzMoreta** [chanzuckerberg/cellxgene-census#1058]
Code:
```python
import cellxgene_census
census = cellxgene_census.open_soma(census_version="2023-07-25")
```

Error: Unable to reach cellxgene servers:
`
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='census.cellxgene.cziscience.com', port=443): Max retries exceeded with url: /cellxgene-census/v1/mirrors.json (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x786c03698fa0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution'))`

Environment in python 3.10 (I have also tried with python 3.9 in case that helped):

```
name: base
channels:
  - conda-forge
  - pytorch
  - defaults
  - nvidia
  - bioconda
dependencies:
  - pyopenssl
  - python=3.9
  - requests
  - pip:
    - -r requirements.txt
  - numpy
  - pytorch::pytorch=2.1=py3.9_cuda12.1*
  - torchvision=>0.16
  - torchaudio
  - biopython
  - h5py
  - scipy
  - scikit-learn
  - pandas
  - matplotlib
  - hdf5plugin
  - pyro-ppl
  - scanpy
  - dill
  - umap-learn
  - seaborn
  - dataframe_image
  - xmltodict
  - nbconvert
  - bioconda::cellxgene
  - conda-forge::lightning
  - numba
  - conda-forge::scvi-tools==1.0.4
  - conda-forge::jax[cuda12_pip]
  - conda-forge::jaxlib
  - conda-forge::huggingface_hub
  - conda-forge::leidenalg
```

requirements.txt

```
gget
cellxgene-census
twine
build
imgkit
```

Perhaps you are doing some server maintenance at the moment? Everything was working until today :)

Thanks!

**Comment by MaximilianLombardo** [comment]
Hi @LysSanzMoreta 

Can you confirm the `cellxgene-census` package version you are using? I tried replicating the error with version 1.9.1 and python 3.9 in a conda environment and didn't encounter the error.

**Comment by LysSanzMoreta** [comment]
Hi! I use micromamba as the package manager and it has installed version `1.10.2`

**Comment by ebezzi** [comment]
Hey @LysSanzMoreta,

this seems to be an issue with your DNS resolver. Normally those are temporary but I'm assuming the issue is persistent for you. 

You should be able to isolate the issue from your Python environment by connecting to that file manually, e.g. via `curl https://census.cellxgene.cziscience.com/cellxgene-census/v1/mirrors.json`. If that also fails, I recommend contacting your IT administrator as they might have added some network restrictions.

**Comment by LysSanzMoreta** [comment]
@ebezzi  Hmm, yeah, it might have been something with the VPN they make me use. I did thought it could be that yesterday, so I deactivated it, nonetheless the problem persisted. 

I did not think about re-starting the computer though (basic IT tip hehhe)...

 Today, it seems to be working again (I have not activated the VPN). So to sum up, if it helps anyone,  do not use the VPN and in case of using the VPN, deactivate it and re-start the computer.

Thanks!
